Sources/SwiftSyntax/Convenience.swift

```diff
@@ -45,3 +45,56 @@ extension MemberAccessExprSyntax {
     )
   }
 }
+
+extension EnumCaseParameterSyntax {
+
+  /// Creates an `EnumCaseParameterSyntax` with a `firstName`, and automatically adds a `colon` to it.
+  ///
+  /// - Parameters:
+  ///   - leadingTrivia: Trivia to be prepended to the leading trivia of the node’s first token. If the node is empty, there is no token to attach the trivia to and the parameter is ignored.
+  ///   - colon: If the parameter has a label, the colon separating the label from the type. SInce `firstName` is not nil, the default `colon` will be set to `TokenSyntax.colonToken()`.
+  ///   - type: The parameter's type.
+  ///   - defaultValue: If the parameter has a default value, the initializer clause describing the default value.
+  ///   - trailingComma: If the parameter is followed by another parameter, the comma separating them.
+  ///   - trailingTrivia: Trivia to be appended to the trailing trivia of the node’s last token. If the node is empty, there is no token to attach the trivia to and the parameter is ignored.
+  public init(
+    leadingTrivia: Trivia? = nil,
+    _ unexpectedBeforeModifiers: UnexpectedNodesSyntax? = nil,
+    modifiers: DeclModifierListSyntax = [],
+    _ unexpectedBetweenModifiersAndFirstName: UnexpectedNodesSyntax? = nil,
+    firstName: TokenSyntax,
+    _ unexpectedBetweenFirstNameAndSecondName: UnexpectedNodesSyntax? = nil,
+    secondName: TokenSyntax? = nil,
+    _ unexpectedBetweenSecondNameAndColon: UnexpectedNodesSyntax? = nil,
+    colon: TokenSyntax = TokenSyntax.colonToken(),
+    _ unexpectedBetweenColonAndType: UnexpectedNodesSyntax? = nil,
+    type: some TypeSyntaxProtocol,
+    _ unexpectedBetweenTypeAndDefaultValue: UnexpectedNodesSyntax? = nil,
+    defaultValue: InitializerClauseSyntax? = nil,
+    _ unexpectedBetweenDefaultValueAndTrailingComma: UnexpectedNodesSyntax? = nil,
+    trailingComma: TokenSyntax? = nil,
+    _ unexpectedAfterTrailingComma: UnexpectedNodesSyntax? = nil,
+    trailingTrivia: Trivia? = nil
+
+  ) {
+      // Wrapping firstName as an Optional so that we can use the generated overload.
+    let firstNameAsOptional: TokenSyntax? = firstName
+    self = .init(leadingTrivia: leadingTrivia,
```

Comment:
Nitpick: I think using `self.init(` in that spot would achieve the same result while also making the code more consistent with other areas in the codebase. 😅
```suggestion
    self.init(leadingTrivia: leadingTrivia,
```